src/app.js

```diff
@@ -7,6 +7,19 @@ export function app(app) {
   var node
   var element
 
+  function hydrate(elm) {
+    return elm == null ? undefined : {
+      tag: elm.tagName,
+      data: {},
+      children: [].map.call(elm.childNodes, function(child){hydrate(child)})
+    }
+  }
+
+  if (root.hasChildNodes !== undefined && root.hasChildNodes()) {
+    element = root.children[0]
+    node = hydrate(element)
+  }
+
```

Comment:
> @andyrj  Is that the expected behavior, if root is not empty then append to children?

The current behavior is to render the tree in the given root or document.body if none is given. 

@ngryman Can you chime in here too? 👋 

> I think it will be clearer if I show a video of the behavior of hyperapp before and after this patch. I am making one now to share for clarification.

Thanks! 👍